### PR TITLE
Update RBACs for tier support in Calico OSS

### PR DESCRIPTION
**Pull request by mazdakn**
## Description

Operator changes needed to support tiers in Calico OSS.
The main PR in Calico: https://github.com/projectcalico/calico/pull/9085

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [x] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: pkg/render/apiserver.go [hunk 1]
```diff
@@ -1309,13 +1311,11 @@ func (c *apiServerComponent) tigeraApiServerClusterRoleBinding() *rbacv1.Cluster
 }
 
 // tierGetterClusterRole creates a clusterrole that gives permissions to get tiers.
-//
-// Calico Enterprise only
 func (c *apiServerComponent) tierGetterClusterRole() *rbacv1.ClusterRole {
 	return &rbacv1.ClusterRole{
 		TypeMeta: metav1.TypeMeta{Kind: "ClusterRole", APIVersion: "rbac.authorization.k8s.io/v1"},
 		ObjectMeta: metav1.ObjectMeta{
-			Name: "tigera-tier-getter",
```
Comment (caseydavenport) [inline pkg/render/apiserver.go:1318]: For enterprise clusters on upgrade, this is going to result in two duplicated ClusterRole objects.

We need some explicit logic to delete the old `tiger-tier-getter` named object to ensure we don't leave unused resources around.

File: pkg/render/apiserver.go [hunk 2]
```diff
@@ -1332,17 +1332,15 @@ func (c *apiServerComponent) tierGetterClusterRole() *rbacv1.ClusterRole {
...
 // get tiers. In k8s 1.15+, cascading resource deletions (for instance pods for a replicaset) failed
 // due to k8s kube-controller not having permissions to get tiers.
-//
-// Calico Enterprise only
 func (c *apiServerComponent) kubeControllerMgrTierGetterClusterRoleBinding() *rbacv1.ClusterRoleBinding {
 	return &rbacv1.ClusterRoleBinding{
 		TypeMeta: metav1.TypeMeta{Kind: "ClusterRoleBinding", APIVersion: "rbac.authorization.k8s.io/v1"},
 		ObjectMeta: metav1.ObjectMeta{
-			Name: "tigera-tier-getter",
+			Name: "calico-tier-getter",
```
Comment (caseydavenport) [inline pkg/render/apiserver.go:1339]: same here (and anywhere we're changing the name of a resource)

File: pkg/render/apiserver.go (hunk 1 above)
Comment (mazdakn) [reply pkg/render/apiserver.go:1318]: Is it better to leave the name unchanged?

File: pkg/render/apiserver.go [hunk 3]
```diff
@@ -1940,5 +1981,23 @@ func (c *apiServerComponent) getDeprecatedResources() []client.Object {
 		},
 	})
 
+	// The following resources were not presetn in Calico OSS, so there is no need to clean up in OSS.
```
Comment (caseydavenport) [inline pkg/render/apiserver.go:1984]: typo: "presetn"

**Review by caseydavenport — APPROVED** [review]
One typo but otherwise LGTM

File: pkg/render/apiserver.go (hunk 3 above)
Comment (mazdakn) [reply pkg/render/apiserver.go]: Fixed.

**Pull request merged**